From 13d02f021c7f96ab1c5f2bc347e1b043d349f9a6 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Mon, 28 Apr 2014 13:27:33 +0800
Subject: [PATCH 2/2] ENGR00306778 - NFS: Upgrading process is shown up when
 NFS boot up. 100%

Previous commit cause /system/core/init build error, since it link libcutils.a
then libselinux.a, so the property_get called in libselinux.a can't resolved.
Use __system_property_get can fix it, since it's in libc.a.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 Android.mk    | 1 -
 src/enabled.c | 4 ++--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/Android.mk b/Android.mk
index 4e9e2b1..6423177 100644
--- a/Android.mk
+++ b/Android.mk
@@ -70,5 +70,4 @@ LOCAL_MODULE_TAGS := eng
 LOCAL_COPY_HEADERS_TO := $(common_COPY_HEADERS_TO)
 LOCAL_COPY_HEADERS := $(common_COPY_HEADERS)
 LOCAL_PRELINK_MODULE := false
-LOCAL_SHARED_LIBRARIES := libcutils
 include $(BUILD_SHARED_LIBRARY)
diff --git a/src/enabled.c b/src/enabled.c
index 6f9b569..f1b73fe 100755
--- a/src/enabled.c
+++ b/src/enabled.c
@@ -7,7 +7,7 @@
 #include <limits.h>
 #include <stdio.h>
 #include "policy.h"
-#include "cutils/properties.h"
+#include <sys/system_properties.h>
 
 int is_selinux_enabled(void)
 {
@@ -19,7 +19,7 @@ int is_selinux_enabled(void)
 	char nfsModeValue[PROP_VALUE_MAX];
 	int ret;
 
-	ret = property_get("ro.nfs.mode", nfsModeValue, NULL);
+	ret = __system_property_get("ro.nfs.mode", nfsModeValue);
 	if((ret ==  3) && (strncmp(nfsModeValue, "yes", 3) == 0)) {
 		return 0;
 	}
-- 
1.8.0

